### ISSUE TITLE
Unable to add lambda trigger to aws_cognito_user_pool due to RecoveryMechanisms error

### ISSUE THREAD

**Issue by cecheta** [hashicorp/terraform-provider-aws#21143]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

* Terraform v1.0.8
* provider registry.terraform.io/hashicorp/aws v3.61.0

### Affected Resource(s)

* aws_cognito_user_pool

### Terraform Configuration Files

```hcl
resource "aws_cognito_user_pool" "pool" {
  name = "mypool"

  lambda_config {
    create_auth_challenge = "arn:aws:lambda:eu-west-2:123456789012:function:my-function"
  }
}

```

### Expected Behavior

```
Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

### Actual Behavior

```
aws_cognito_user_pool.pool: Modifying... [id=eu-west-2_aaaaaaaaa]
╷
│ Error: error updating Cognito User pool (eu-west-2_aaaaaaaaa): InvalidParameter: 1 validation error(s) found.
│ - minimum field size of 1, UpdateUserPoolInput.AccountRecoverySetting.RecoveryMechanisms.
│ 
│ 
│   with aws_cognito_user_pool.pool,
│   on main.tf line 1, in resource "aws_cognito_user_pool" "pool":
│    1: resource "aws_cognito_user_pool" "pool" {
│ 
╵
```

### Steps to Reproduce

1. Create a `aws_cognito_user_pool` resource with Terraform

```hcl
resource "aws_cognito_user_pool" "pool" {
  name = "mypool"
}
```

2. In the AWS Cognito console, under 'MFA and verifications', change the 'How will a user be able to recover their account?' setting to a different value (i.e. Email if available, otherwise phone, but don’t allow a user to reset their password via phone if they are also using it for MFA). Save changes.

![image](https://user-images.githubusercontent.com/60179183/135915118-b22858fb-3e3f-4123-9729-76f21932c672.png)

3. Again, under 'MFA and verifications', change the 'How will a user be able to recover their account?' setting back to '(Not Recommended) Phone if available, otherwise email, and do allow a user to reset their password via phone if they are also using it for MFA.'. Save changes.

![image](https://user-images.githubusercontent.com/60179183/135915210-bcf1d714-2940-4986-8872-1f6c292f42e8.png)

4. Add a `lambda_config` block to the `aws_cognito_user_pool` resource, with a lambda trigger.

```hcl
resource "aws_cognito_user_pool" "pool" {
  name = "mypool"

  lambda_config {
    create_auth_challenge = "arn:aws:lambda:eu-west-2:012345678901:function:my-function"
  }
}
```

5. `terraform apply`

### Further Notes

It is worth nothing that after initially creating the user pool, the response from the AWS CLI command `describe-user-pool` does not contain the property `AccountRecoverySetting`, and in the Terraform state file, `account_recovery_setting` is equal to `[]`. However, after completing steps 2 and 3, the AWS CLI response now contains `"AccountRecoverySetting": {}`, and the Terraform state file now contains:
```json
"account_recovery_setting": [
    {
        "recovery_mechanism": []
    }
]
```

**Comment by jakemillerkasasa** [comment]
Hello, do we have an update on when this bug will be looked at?

**Comment by jakemillerkasasa** [comment]
Hello, do we have insight on when this might get merged and if there is a targeted release to expect to see this change in? Thanks!

**Comment by nicolaei** [comment]
For anyone looking for a quick-fix while waiting for this to be merged:

The latest version of v3 I found working is `3.18.0`. The regression is seemingly caused by 78b2e4a, which is included in `3.19.0`.

At least this worked when trying to upgrade an old repo from v2 to v3 for me today.
